enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/QueryId.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.enterprise.builtinprocs;
+
+public final class QueryId
+{
+    public static final String QUERY_ID_PREFIX = "query-";
+    private final long kernelQueryId;
+
+    public static QueryId queryId( long queryId )
+    {
+        return new QueryId( queryId );
+    }
+
+    public static QueryId parseQueryId( String queryIdText )
+    {
+        try
+        {
+            if ( queryIdText.startsWith( QUERY_ID_PREFIX ) )
+            {
+                return new QueryId( Long.parseLong( queryIdText.substring( QUERY_ID_PREFIX.length() ) ) );
+            }
+        }
+        catch ( NumberFormatException e )
+        {
+            throw new IllegalArgumentException( e );
```

Comment:
This exception missing error details and expected format. It would be strange to get illegal argument exception without explanation what format is actually expected from you